activerecord/test/cases/associations_test.rb

```diff
@@ -322,6 +322,20 @@ def test_target_merging_recognizes_updated_in_memory_records
 
     assert_not_empty member.favorite_memberships.to_a
   end
+
+  def test_size_differentiates_between_new_and_persisted_in_memory_records_when_loaded_records_are_empty
+    member = members(:blarpy_winkup)
+    assert_empty member.favorite_memberships
+
+    membership = member.favorite_memberships.create!
+    membership.update!(favorite: false)
+
+    # CollectionAssociation#size has different behavior when loaded vs. non-loaded
+    # the first call will mark the association as loaded and the second call will
+    # take a different code path, so it's important to keep both assertions
+    assert_equal 0, member.favorite_memberships.size
+    assert_equal 0, member.favorite_memberships.size
```

Comment:
I'm having a hard time reasoning about why the size of `member.favorite_memberships.size` is 0. The count of `member.favorite_memberships.count` is 1, right? Shouldn't the addition of a persisted record to the relation therefore change the size to 1 as well?  we're coming across this case where we are surprised that the following is now true

```ruby
    membership = member.favorite_memberships.create!
    refute_included membership, member.favorite_memberships
```